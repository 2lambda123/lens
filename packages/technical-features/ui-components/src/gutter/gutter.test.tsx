import React from "react";

import { Gutter, GutterProps } from "./Gutter";
import { render } from "@testing-library/react";

interface Scenario {
  name: string;
  props: GutterProps;
}

describe("Gutter", () => {
  (
    [
      { name: "when size is not set", props: {} },
      { name: "when size is sm", props: { size: "sm" } },
      { name: "when size is md", props: { size: "md" } },
      { name: "when size is xl", props: { size: "xl" } },
    ] as Scenario[]
  ).forEach((scenario) => {
    it(`"${scenario.name}", renders`, () => {
      const rendered = render(<Gutter {...scenario.props} />);

      expect(rendered.baseElement).toMatchSnapshot();
    });
  });
});
