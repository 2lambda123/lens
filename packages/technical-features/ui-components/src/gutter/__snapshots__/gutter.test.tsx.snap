// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gutter "when size is md", renders 1`] = `
<body>
  <div>
    <div
      class="w-6 h-6"
    />
  </div>
</body>
`;

exports[`Gutter "when size is not set", renders 1`] = `
<body>
  <div>
    <div
      class="w-6 h-6"
    />
  </div>
</body>
`;

exports[`Gutter "when size is sm", renders 1`] = `
<body>
  <div>
    <div
      class="w-4 h-4"
    />
  </div>
</body>
`;

exports[`Gutter "when size is xl", renders 1`] = `
<body>
  <div>
    <div
      class="w-10 h-10"
    />
  </div>
</body>
`;
